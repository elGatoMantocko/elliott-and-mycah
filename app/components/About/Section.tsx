import { Box, Typography } from '@mui/material';
import * as React from 'react';

import { ScriptTypography } from '../ScriptTypography';

type AboutSectionProps = {
  /**
   * Section title displayed above body text.
   */
  title: string | JSX.Element;
  /**
   * Section body displayed to the left or right.
   */
  body: string | JSX.Element;
  /**
   * Image to be displayed next to the copy.
   */
  imageSource?: string;
  /**
   * Image alt text.
   */
  imageAlt?: string;
};

export const Section = ({ title, body, imageSource, imageAlt }: AboutSectionProps) => (
  <Box display="flex" flexDirection="column" pb={1} maxWidth={470}>
    <Box mx={1} px={1} mt={3}>
      <ScriptTypography color="secondary" variant="h3" align="center">
        {title}
      </ScriptTypography>
      <Typography gutterBottom align="center">
        {body}
      </Typography>
    </Box>
    {imageSource && (
      <Box mt={2} mx="auto" px={1} maxWidth={400}>
        <img
          style={{ maxWidth: '100%', maxHeight: '60vh', borderRadius: '1rem' }}
          src={imageSource}
          alt={imageAlt}
        />
      </Box>
    )}
  </Box>
);
