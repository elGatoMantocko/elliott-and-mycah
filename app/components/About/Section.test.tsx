import { render } from '@testing-library/react';
import React from 'react';

import { Section } from './Section';

it('should render a <Section />', async () => {
  const el = render(
    <div data-testid="test-section">
      <Section title="Hello world" body="This is a longer part of the section" />,
    </div>,
  );
  expect(await el.findByTestId('test-section')).toMatchSnapshot('basic-section');
});
