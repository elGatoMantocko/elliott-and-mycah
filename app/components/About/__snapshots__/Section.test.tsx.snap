// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a <Section /> > basic-section 1`] = `
<div
  data-testid="test-section"
>
  <div
    class="MuiBox-root css-1cgxlb6"
  >
    <div
      class="MuiBox-root css-s3cama"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-npk0zx-MuiTypography-root"
      >
        Hello world
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter MuiTypography-gutterBottom css-1c3pd2j-MuiTypography-root"
      >
        This is a longer part of the section
      </p>
    </div>
  </div>
  ,
</div>
`;
